src/mpsc/errors.rs

```diff
@@ -88,6 +110,72 @@ impl<T> fmt::Display for Closed<T> {
 #[cfg(feature = "std")]
 impl<T> std::error::Error for Closed<T> {}
 
+// === impl SendTimeoutError ===
+
+#[cfg(feature = "std")]
+impl SendTimeoutError {
+    pub(crate) fn with_value<T>(self, value: T) -> SendTimeoutError<T> {
+        match self {
+            Self::Timeout(()) => SendTimeoutError::Timeout(value),
+            Self::Closed(()) => SendTimeoutError::Closed(value),
+        }
+    }
+
```

Comment:
gah, my bad, i guess we had two impl blocks here for a reason...
```suggestion
}

#[cfg(feature = "std")]
impl<T> SendTimeoutError<T> {
```